src/development/plugins-local/src/github.com/PatrickMi/body-forward-auth/body_forward_auth.go

```diff
@@ -0,0 +1,86 @@
+package body_forward_auth
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httputil"
+)
+
+// Config holds configuration to be passed to the plugin i.e. variables that come from the dockerfile
+type Config struct {
+	AuthUrl string
+}
+
+// CreateConfig populates the Config data object
+func CreateConfig() *Config {
+	return &Config{
+		// Default values can be set here like
+		// AuthUrl: "http://127.0.0.1",
+	}
+}
+
+type BodyForwardAuth struct {
+	next    http.Handler
```

Comment:
Good question. After first agreeing with you on the tab point I tried to setup the go formatter in VSCode and had some issues. After a short google search I stumbled upon this: https://go.dev/doc/effective_go#formatting. According to this document go has only a few rules regarding formatting. One of them is that go uses tabs instead of spaces for indentation. This formatting rule should be used if one wants to use the official go formatter as it is not a good style to change the formatting (as described here: https://github.com/microsoft/vscode-go/issues/2871#issuecomment-564833962). 

There is a package that can be installed to use a stricter go formatting ruleset (https://github.com/mvdan/gofumpt). However this also has no rule that enforces a blank line after a code block.

As I still agree with you on the point regarding the blank line after a block of code I added them by hand in the file.